src/Task.ts

```diff
@@ -684,6 +683,99 @@ export class Task {
         return linkText;
     }
 
+    /**
+     * Compare two lists of Task objects, and report whether their
+     * tasks are identical in the same order.
+     *
+     * This can be useful for optimising code if it is guaranteed that
+     * there are no possible differences in the tasks in a file
+     * after an edit, for example.
+     *
+     * If any field is different in any task, it will return false.
+     *
+     * @param oldTasks
+     * @param newTasks
+     */
+    static tasksListsIdentical(oldTasks: Task[], newTasks: Task[]): boolean {
+        if (oldTasks.length !== newTasks.length) {
+            return false;
+        }
+        return oldTasks.every((oldTask, index) =>
+            oldTask.identicalTo(newTasks[index]),
+        );
+    }
+
+    /**
+     * Compare all the fields in another Task, to detect any differences from this one.
+     *
+     * If any field is different in any way, it will return false.
+     *
+     * This is used in some optimisations, to avoid work if an edit to file
+     * does not change any tasks, so it is vital that its definition
+     * of identical is very strict.
+     *
+     * @param other
+     */
+    public identicalTo(other: Task) {
+        // Based on ideas from koala. AquaCat and javalent in Discord:
+        // https://discord.com/channels/686053708261228577/840286264964022302/996735200388186182
+        // and later.
+        let args: Array<keyof Task> = [
+            'status',
+            'description',
+            'path',
+            'indentation',
+            'sectionStart',
+            'sectionIndex',
+            'originalStatusCharacter',
+            'precedingHeader',
+            'priority',
+            'blockLink',
+        ];
+        for (const el of args) {
+            if (this[el] !== other[el]) return false;
+        }
+
+        // Compare tags
+        if (this.tags.length !== other.tags.length) {
+            return false;
+        }
+        // Tags are the same only if the values are in the same order
+        if (
+            !this.tags.every(function (element, index) {
+                return element === other.tags[index];
+            })
+        ) {
+            return false;
+        }
+
+        // Compare Date fields
+        args = ['startDate', 'scheduledDate', 'dueDate', 'doneDate'];
+        for (const el of args) {
+            const date1 = this[el] as Moment | null;
+            const date2 = other[el] as Moment | null;
+            if (Sort.compareByDate(date1, date2) !== 0) {
+                return false;
+            }
+        }
+
+        const recurrence1 = this.recurrence;
+        const recurrence2 = other.recurrence;
+        if (recurrence1 === null && recurrence2 !== null) {
+            return false;
+        } else if (recurrence1 !== null && recurrence2 === null) {
+            return false;
```

Comment:
I'm a bit uncomfortable with this, but don't yet know how to write generic code that check of equality of `'Thing | null'`.

Reply:
Am not going to change this now, as the change is well tested with this code and working fine.